storage/decorator.py

```diff
@@ -3,75 +3,63 @@
 # you may not use this file except in compliance with the Elastic License 2.0.
 
 import zlib
-from typing import Any, Callable, Iterator
+from typing import Any, Iterator
 
 from share import shared_logger
 
-from .storage import CommonStorageType
+from .storage import CommonStorageType, GetByLinesCallable
 
 
-def by_lines(
-    func: Callable[[CommonStorageType, int, Any, str, int], Iterator[tuple[bytes, int, int]]]
-) -> Callable[[CommonStorageType, int, Any, str, int], Iterator[tuple[bytes, int, int]]]:
+def by_lines(func: GetByLinesCallable[CommonStorageType]) -> GetByLinesCallable[CommonStorageType]:
     def wrapper(
         storage: CommonStorageType, range_start: int, body: Any, content_type: str, content_length: int
     ) -> Iterator[tuple[bytes, int, int]]:
+        offset: int = range_start
         unfinished_line: bytes = b""
-        _offset: int = 0
-        _range_start: int = 0
 
-        _newline: bytes = b""
-        _newline_length: int = 0
+        newline: bytes = b""
+        newline_length: int = 0
         iterator = func(storage, range_start, body, content_type, content_length)
-        while True:
-            try:
-                data, range_start, newline_length = next(iterator)
-                if _offset == 0:
-                    _offset = _range_start
-
-                unfinished_line = unfinished_line + data
-                lines = unfinished_line.decode("UTF-8").splitlines()
-
-                if len(lines) > 0:
-                    if _newline_length == 0:
-                        if unfinished_line.find(b"\r\n") > 0:
-                            _newline = b"\r\n"
-                            _newline_length = len(_newline)
-                        else:
-                            _newline = b"\n"
-                            _newline_length = len(_newline)
-
-                    if unfinished_line.endswith(_newline):
-                        unfinished_line = lines.pop().encode() + _newline
+        for data, range_start, _newline_length in iterator:
+            unfinished_line = unfinished_line + data
+            lines = unfinished_line.decode("UTF-8").splitlines()
+
+            if len(lines) > 0:
```

Comment:
Nit. If there is nothing to do if there are no lines, maybe you can `continue` directly here. Saving in nesting levels uses to be a good idea in Python :slightly_smiling_face: 
```suggestion
            if len(lines) == 0:
                continue
```